components/Poll/Poll.jsx

```diff
@@ -0,0 +1,342 @@
+          if (i === optIndex) {
+            this.updateOptionVoters(opts[optIndex]);
+          } else {
+            let userIndex = this.userVoteIndex(opts[i]);
+            if (userIndex >= 0) {
+              opts[i].usersVoted.splice(userIndex, 1);
+            }
+          }
+        }
+      }
+    }
+    this.props.savePoll && this.props.savePoll({
+      title: this.state.title,
+      multiSelect: this.state.multiSelect,
+      open: this.state.open,
+      options: opts,
+    });
+  }
+
+  confirmDelete = async (item) => {
+    if (!this.props.isAuthor || !this.props.savePoll) {
+      return;
+    }
+    Alert.alert(
+      'Hold Up!',
+      `Are you sure you want to remove this option?\nOption: ${item.text}`,
+      [
+        { text: 'Cancel', style: 'cancel' },
+        { text: 'Delete', onPress: this.deleteOption.bind(this, item) },
+      ],
+    );
+  }
+
+  deleteOption = (item) => {
+    var opts = this.state.options;
+    var optIndex = opts.findIndex(opt => opt.key === item.key);
+    if (optIndex === -1) {
+      return;
+    }
+    opts.splice(optIndex, 1);
+    this.props.savePoll({
+      title: this.state.title,
+      multiSelect: this.state.multiSelect,
+      open: this.state.open,
+      options: opts,
+    });
+  }
+
+  addOption = async () => {
+    if (this.state.newOption.length == 0) {
+      return;
+    }
+    let optionText = this.state.newOption;
+    this.setState({ newOption: '' });
+    if (!this.props.savePoll) {
+      pollOption(this.props.postId, { option: optionText })
+      .then((poll) => {
+        this.setState({ options: poll.options });
+        this.props.updatePoll && this.props.updatePoll({
+          title: this.state.title,
+          multiSelect: this.state.multiSelect,
+          open: this.state.open,
+          options: poll.options,
+        });
+      })
+      .catch((error) => alert("Error adding poll option. Sorry about that!"));
+    } else {
+      var opts = this.state.options;
+      opts.push({
+        key: `${this.state.options.length}`,
+        text: optionText,
+        usersVoted: [],
+      });
+      this.props.savePoll({
+        title: this.state.title,
+        multiSelect: this.state.multiSelect,
+        open: this.state.open,
+        options: opts,
+      });
+    }
+  }
+
+  buildListItems = () => {
+    let items = this.state.options
+            .sort((a, b) => b.usersVoted.length - a.usersVoted.length)
+            .slice()
+            .map(item => {
+              if (item._id) {
+                item.key = item._id;
+              }
+              return item;
+            });
+    if (!!this.props.isAuthor || this.state.open) {
+      items.push({ key: 'add_option_item' });
+    }
+    return items;
+  }
+
+  renderListItem = ({ item }) => {
+    if (item.key === 'add_option_item') {
+      return (
+        <View style={styles.optionGroup}>
+          <TouchableOpacity style={styles.addContainingView} onPress={this.addOption}>
```

Comment:
The touchable opacity is too small, it seems to be really easy to accidentally close the editing view when I try to add new options :(

Reply:
I added some more padding in the most recent update, please tell me if seems good to you